blockchain/processing.py

```diff
@@ -429,6 +429,7 @@ def _execute_phase_3(self, config, phase_2_info):
             # storing valid verification record
             verification_db.insert_verification(phase_2_record)
 
+            # retrieve all phase 2 records for current block
             phase_2_records = self.get_sig_records(phase_2_record)
```

Comment:
I'd move this line below 438, that way we only fetch the p2 records when the reqs are met and they're needed for the transfer records.